Send correct status header for Outbox items

pfefferle opened the pull request:
I had issues on several servers that the WordPress 404 page was not loaded properly but a webserver error message instead. This PR fixes the issue and adds a proper 406 code when the Accept type is wrong.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Send a HTTP "406 Not Acceptable" error code if a server queries an Outbox Item with a missing `application/json+activity` Accept header.
* It also adds the `404` query var, to ensure that the 404 page will be loaded in the frontend.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [X] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [X] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Better error handling when other servers request Outbox items in the wrong format, and 404 pages now show correctly.

</details>

pfefferle committed:
Send correct status header for Outbox items

Send a HTTP "406 Not Acceptable" error code if a server queries an Outbox Item with a missing `application/json+activity` Accept header.

It also adds the 404 flag, to ensure that the 404 page will be loaded in the frontend.

matticbot committed:
Add changelog

obenland commented:
Is this at all unit-testable? It would be nice to have that covered.

obenland commented on includes/class-activitypub.php:
```diff
@@ -141,6 +141,10 @@ public static function render_activitypub_template( $template ) {
 		self::add_headers();
 
 		if ( ! is_activitypub_request() || ! should_negotiate_content() ) {
+			if ( \get_query_var( 'p' ) && Outbox::POST_TYPE === \get_post_type( \get_query_var( 'p' ) ) ) {
```
```suggestion
			if ( Outbox::POST_TYPE === \get_query_var( 'post_type' ) ) {
```
Does this work? For custom post types, it should always define the `post_type` var, otherwise it defaults to `post`.

pfefferle replied:
Sadly not! That was what I tried first ☺️

I would assume that it does not work because the post type is not public!?!

pfefferle committed:
Merge branch 'trunk' into improve/error-codes

pfefferle committed:
Add unittests

pfefferle committed:
Merge branch 'trunk' into improve/error-codes

obenland committed:
Create proper Outbox item

Makes test pass when calling directly (`npm run env-test -- --filter test_outbox_post_type_non_activitypub_request_returns_406`)

obenland committed:
Merge branch 'trunk' into improve/error-codes
